app/queries/articles/active_threads_query.rb

```diff
@@ -8,23 +8,25 @@ class ActiveThreadsQuery
 
     MINIMUM_SCORE = -4
 
-    def self.call(relation: Article.published, **options)
+    def self.call(relation: Article.published, options: {})
       options = DEFAULT_OPTIONS.merge(options)
       tags, time_ago, count = options.values_at(:tags, :time_ago, :count)
 
       relation = relation.limit(count)
+      relation = tags.size == 1 ? relation.cached_tagged_with(tags.first) : relation.tagged_with(tags)
       relation = if time_ago == "latest"
                    relation = relation.where(score: MINIMUM_SCORE..).presence || relation
                    relation.order(published_at: :desc)
                  elsif time_ago
                    relation = relation.where(published_at: time_ago.., score: MINIMUM_SCORE..).presence || relation
                    relation.order(comments_count: :desc)
+                 elsif tags.present?
```

Comment:
Given we always expect tags to be present, that was an assumption that I missed before when looking at this whole query threads class, we can remove this `if tags.present?` and leave it as simply an `else`

Reply:
I made this adjustment prior to seeing this message! 😄